### ISSUE TITLE
Validator not working for array fields

### ISSUE THREAD

**Issue by rafaferry** [deepkit/deepkit-framework#53]
Hello,

I'm having some problem with validator not being invoked. 

Here's a test case:

```
import {
  f,
  PropertyValidatorError,
  PropertyValidator,
  validatedPlainToClass,
  validate,
  plainToClass,
} from '@marcj/marshal';

import { expect } from 'chai';

class NotEmptyArrayValidator implements PropertyValidator {
  validate<T>(value: []): PropertyValidatorError | void {
    console.log('INSIDE VALIDATOR');
    if (value === null || value.length === 0) {
      return new PropertyValidatorError('notEmpty', 'notEmpty');
    }
  }
}

enum Role {
  ADMIN = 'ADMIN',
}

class TestEntity {
  @(f
    .enum(Role, true)
    .asArray()
    .validator(NotEmptyArrayValidator))
  roles: Role[] = [];
}

describe('Empty Array Validator', () => {
  it('should throw error when validatedPlainToClass', async () => {
    const testEntity = validatedPlainToClass(TestEntity, {});
    expect.fail('should have thrown exception');
  });
  it('should return error when validate', function() {
    const testEntity = plainToClass(TestEntity, {});
    const validationResult = validate(TestEntity, testEntity);
    expect(validationResult.length).to.be.equals(1);
  });
});

```
This test case fail in both "expect"

Am I doing something wrong?

**Comment by marcj** [comment]
Indeed, currently array and map types do not have validators or meta data attached. The definition directly redirects to its content type. Funny timing, because I refactored Marshal in the last days to support exactly that, which is already done. The core would then look like that:

```
class TestEntity {
  @f.array(f.enum(Role, true)).validator(NotEmptyArrayValidator))
  roles: Role[] = [];
}
```

When I added the ORM stuff and finalize the new API I'll publish a new release, but under the `@super-hornet/marshal` package name. I will ping you once done.
